x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java

```diff
@@ -44,7 +54,78 @@
             null
         ));
 
-    static final Map<String, ServiceAccount> ACCOUNTS = List.of(FLEET_ACCOUNT).stream()
+    private static final ServiceAccount KIBANA_ACCOUNT = new ElasticServiceAccount("kibana-system",
+        new RoleDescriptor(
+            NAMESPACE + "/kibana-system",
+            new String[] {
+                "monitor", "manage_index_templates", MonitoringBulkAction.NAME, "manage_saml", "manage_token", "manage_oidc",
+                InvalidateApiKeyAction.NAME, "grant_api_key",
+                GetBuiltinPrivilegesAction.NAME, "delegate_pki", GetLifecycleAction.NAME,  PutLifecycleAction.NAME,
+                // To facilitate ML UI functionality being controlled using Kibana security privileges
+                "manage_ml",
+                // The symbolic constant for this one is in SecurityActionMapper, so not accessible from X-Pack core
+                "cluster:admin/analyze",
+                // To facilitate using the file uploader functionality
+                "monitor_text_structure",
+                // To cancel tasks and delete async searches
+                "cancel_task"
+            },
+            new RoleDescriptor.IndicesPrivileges[] {
```

Comment:
**question** is there a way to share this role descriptor with the one used to define the `kibana_system` reserved role, so that we have a single source of truth for our necessary privileges?

Reply:
Glad to hear the news!
We'll push for a single source of truth for the reserved role and the service account, I don't think there's any issue preventing this.